grpclog/grpclog.go

```diff
@@ -0,0 +1,441 @@
+package grpclog
```

Comment:
Is this essentially because we need to vendor pu/pu's logging strategy? Would making [this](https://github.com/pulumi/pulumi/blob/035a502d86403d815059615a9c047ccccc2cbdd5/pkg/util/rpcdebug/logformat.go#L24) public help?

Reply:
That might help somewhat, but I'm also interested in parsing the log messages back into their original request and response types so we can access all properties within them.